src/main/java/uk/ac/ic/wlgitbridge/bridge/lock/InProcessLock.java

```diff
@@ -0,0 +1,48 @@
+package uk.ac.ic.wlgitbridge.bridge.lock;
+
+import uk.ac.ic.wlgitbridge.util.Log;
+
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * A thin wrapper around a ReentrantLock
+ */
+public class InProcessLock implements ProjectLock {
+  private String projectName;
+  private ReentrantLock lock;
+  private OperationTracker registry;
+
+  public InProcessLock(String projectName, OperationTracker operationTracker) {
+    this.projectName = projectName;
+    this.registry = operationTracker;
+    this.lock = new ReentrantLock();
```

Comment:
Review: We add `this` to the static map, keyed on projectName, then increment the refcount.

e.g.
```Java
// may have to wrap this in a synchronized function
if (!InProcessLock.hashMap.contains(projectName)) {
  InProcessLock.hashMap.insert(projectName, makePair(this, 0));
}

// such pseudocode
InProcessLock.hashMap[projectName].integerValue ++;
```